src/ocean/util/log/model/ILogger.d

```diff
@@ -96,52 +205,94 @@ interface ILogger
+    ***************************************************************************/
 
-                Is this logger additive? That is, should we walk ancestors
-                looking for more appenders?
+    public bool additive ();
 
-        ***********************************************************************/
+    /***************************************************************************
 
-        bool additive ();
+        Set the additive status of this logger
 
-        /***********************************************************************
+        Additive loggers walk through ancestors looking for more appenders
 
-                Set the additive status of this logger. See isAdditive().
+        Params:
+            enabled = Whereas this logger is additive.
 
-        ***********************************************************************/
+        Returns:
+            `this` for easy chaining
 
-        ILogger additive (bool enabled);
+    ***************************************************************************/
 
-        /***********************************************************************
+    public ILogger additive (bool enabled);
 
-                Send a message to this logger via its appender list.
+    /***************************************************************************
 
-        ***********************************************************************/
+        Send a message to this logger.
+
+        Params:
+            level = Level at which to log the message
+            exp   = Lazily evaluated message string
+                    If the `level` is not enabled for this logger, it won't
+                    be evaluated.
 
-        ILogger append (Level level, lazy cstring exp);
+        Returns:
+            `this` for easy chaining
+
+    ***************************************************************************/
+
+    public ILogger append (Level level, lazy cstring exp);
+}
+
+unittest
+{
+    test!("==")(ILogger.convert(ILogger.Level.Trace), "Trace");
+    test!("==")(ILogger.convert(ILogger.Level.Info), "Info");
+    test!("==")(ILogger.convert(ILogger.Level.Warn), "Warn");
+    test!("==")(ILogger.convert(ILogger.Level.Error), "Error");
+    test!("==")(ILogger.convert(ILogger.Level.Fatal), "Fatal");
+    test!("==")(ILogger.convert(ILogger.Level.None), "None");
+}
+
+unittest
+{
+    test!("==")(ILogger.convert("info"), ILogger.Level.Info);
+    test!("==")(ILogger.convert("Info"), ILogger.Level.Info);
+    test!("==")(ILogger.convert("INFO"), ILogger.Level.Info);
+    test!("==")(ILogger.convert("FATAL"), ILogger.Level.Fatal);
+    // Use the default value
+    test!("==")(ILogger.convert("Info!"), ILogger.Level.Trace);
+    test!("==")(ILogger.convert("Baguette", ILogger.Level.Warn),
+                ILogger.Level.Warn);
+    // The first entry in the array
+    test!("==")(ILogger.convert("trace", ILogger.Level.Error),
+                ILogger.Level.Trace);
```

Comment:
Those unittests are already present in the new logger (as well as the method pasted here).